dask/array/slicing.py

```diff
@@ -526,9 +557,18 @@ def take(outname, inname, blockdims, index, axis=0):
     {('y', 0): (getitem, ('x', 0), ([1, 3, 5],)),
      ('y', 2): (getitem, ('x', 2), ([7],))}
     """
-    if issorted(index):
```

Comment:
For the record, Numpy's sort is a quicksort by default, so will never get below O(n log n). Python's sort is a "timsort" (you can look it up if you're curious, it's quite sophisticated) which is able to achieve O(n) on already sorted arrays, but with a much higher multiplicative constant.